spylib/store.py

```diff
@@ -138,16 +145,28 @@ async def shoprequest(self, goodstatus, debug, endpoint, **kwargs) -> Dict[str,
         stop=stop_after_attempt(5),
         retry=retry_if_exception_type(ShopifyThrottledError),
     )
-    async def execute_gql(self, query: str, variables: Dict[str, Any] = {}) -> Dict[str, Any]:
+    async def execute_gql(
+        self, query: str, variables: Dict[str, Any] = {}, operation_name: Optional[str] = None
+    ) -> Dict[str, Any]:
         """Simple graphql query executor because python has no decent graphql client"""
 
         url = f'{self.url}/graphql.json'
         headers = {
             'Content-type': 'application/json',
             'X-Shopify-Access-Token': self.access_token,
         }
+        body = {
+            'query': query,
+            'variables': variables,
+        }
+
+        if operation_name:
+            body['operationName'] = operation_name
```

Comment:
I _think_ that works:
```suggestion
            'variables': variables,
            'operationName': operation_name,
        }
```

Reply:
It should work, I was worried about it throwing this error which happens when you pass in `operationName=Null`:
![image](https://user-images.githubusercontent.com/12765521/128055439-a53493a8-62b9-49a8-b8f7-477ea9063a80.png)
But I looked closer and it only happens if you have more than 1 named query.